File: Week 5/bruteforce_caesar.py
def caesar_decrypt(ciphertext, shift):
    decrypted = ''
    for char in ciphertext:
        if char.isalpha():
            base = ord('A') if char.isupper() else ord('a')
            decrypted += chr((ord(char) - base - shift) % 26 + base)
        else:
            decrypted += char
    return decrypted

def brute_force_caesar(ciphertext):
    print("Brute Force Attack Results:")
    for key in range(1, 26):
        attempt = caesar_decrypt(ciphertext, key)
        print(f"Key {key:2}: {attempt}")

# Example Ciphertext (encrypted "THIS IS A SECRET")
ciphertext = "Wklv lv d vhfuhw"
brute_force_caesar(ciphertext)

File: Week 5/bruteforce_caesar.cpp
#include <iostream>
#include <string>
using namespace std;

// Decrypt Caesar cipher with given shift
string caesarDecrypt(string ciphertext, int shift) {
    string decrypted = "";
    for (char c : ciphertext) {
        if (isalpha(c)) {
            char base = isupper(c) ? 'A' : 'a';
            decrypted += (char)(((c - base - shift + 26) % 26) + base);
        } else {
            decrypted += c;
        }
    }
    return decrypted;
}

int main() {
    string ciphertext = "Wklv lv d vhfuhw"; // "This is a secret" with key 3

    cout << "Brute Force Caesar Attack:\n";
    for (int key = 1; key < 26; ++key) {
        string decrypted = caesarDecrypt(ciphertext, key);
        cout << "Key " << (key < 10 ? " " : "") << key << ": " << decrypted << endl;
    }

    return 0;
}
